presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -50,6 +50,10 @@
 
     boolean testBytes(byte[] buffer, int offset, int length);
 
+    int getPrecedingPositionsToFail();
```

Comment:
@highker Added the following comments:

```
    /**
     * A filter becomes non-deterministic when applies to nested column,
     * e.g. a[1] > 10 is non-deterministic because > 10 filter applies only to some
     * positions, e.g. first entry in a set of entries that correspond to a single
     * top-level position.
     */
    boolean isDeterministic();

    boolean testNull();

    /**
     * Used to apply is [not] null filters to complex types, e.g.
     * a[1] is null AND a[3] is not null, where a is an array(array(T)).
     *
     * In these case, the exact values are not known, but it is known whether they are
     * null or not. Furthermore, for some positions only nulls are allowed (a[1] is null),
     * for others only non-nulls (a[3] is not null), and for the rest both are allowed
     * (a[2] and a[N], where N > 3).
     */
    boolean testNonNull();

    boolean testLong(long value);

    boolean testDouble(double value);

    boolean testFloat(float value);

    boolean testDecimal(long low, long high);

    boolean testBoolean(boolean value);

    boolean testBytes(byte[] buffer, int offset, int length);

    /**
     * When a filter applied to a nested column fails, the whole top-level position should
     * fail. To enable this functionality, the filter keeps track of the boundaries of
     * top-level positions and allows the caller to find out where the current top-level
     * position started and how far it continues.
     * @return number of positions from the start of the current top-level position up to
     * the current position (excluding current position)
     */
    int getPrecedingPositionsToFail();

    /**
     * @return number of positions remaining until the end of the current top-level position
     */
    int getSucceedingPositionsToFail();
```